src/main/java/seedu/programmer/logic/commands/AddLabCommand.java

```diff
@@ -47,11 +48,17 @@ public AddLabCommand(Lab result) {
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
 
-        // Gets the last filtered list displayed
-        List<Student> lastShownList = model.getFilteredStudentList();
         if (result.getLabTotal().getLabTotalScore() < 0) {
             throw new CommandException(LAB_SCORE_MESSAGE_CONSTRAINTS);
         }
+
+        // Gets the last filtered list displayed
+        List<Student> lastShownList = model.getFilteredStudentList();
```

Comment:
Doesn't this get the last displayed list instead of all the students in the model?

Reply:
This is solved in #452